### PR TITLE
Feature/make angular15 compatible

### PR DESCRIPTION
**Pull request by dsteegen**
This PR will make the library compatible with Angular 15. The changes are inspired by this [fork](https://github.com/fast-facts/ngx-bootstrap-multiselect).

### CODE REVIEW

**Comment by sommmen** [comment]
Shouldn't the deps like 

```
@angular/core": "15.2.0
```

Be ^15.0.0?

**Comment by dsteegen** [comment]
> Shouldn't the deps like
> 
> ```
> @angular/core": "15.2.0
> ```
> 
> Be ^15.0.0?

Yes! That is indeed better than fixating it to 15.2.0. I already updated it in this PR.

Most changes were taken over from another fork so if you spot other things that can be improved; let me know :)

**Comment by sommmen** [comment]
> > Shouldn't the deps like
> > ```
> > @angular/core": "15.2.0
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Be ^15.0.0?
> 
> Yes! That is indeed better than fixating it to 15.2.0. I already updated it in this PR.
> 
> Most changes were taken over from another fork so if you spot other things that can be improved; let me know :)

I'll take a look tomorrow, am on phone atm.

**Comment by dsteegen** [comment]
> > 
> 
> Uh also, I'm not a js dev, is it easy to create a beta package I can test with my app? We kind of use all features extensively so testing the pr would be nice for me.

You can build it manually if you clone my repository and switch to my branch. Afterwards you have to execute the following steps:

- npm install in the root folder
- npm run build
- in the dist folder, run npm pack. This should create ngx-bootstrap-multiselect-3.0.0.tgz inside the dist folder
- in your project's package.json, replace the ngx-bootstrap-multiselect entry with `"ngx-bootstrap-multiselect": "/Users/davysteegen/ngx-bootstrap-multiselect/dist/ngx-bootstrap-multiselect-3.0.0.tgz",` (change the directory to match your own)
- run npm install on your project's package.json.

You should now have the version that corresponds to this PR.

**Comment by dsteegen** [comment]
@sommmen 
FYI: if you don't want to build a local version yourself, your can always use a published version from https://www.npmjs.com/package/ngx-bootstrap-multiselect-updated

The fork/PR I created is based on that NPM package (which is also a fork from the original ngx-bootstrap-multiselect from @softsimon). I hope this helps.

File: package.json [hunk 1]
```diff
@@ -12,12 +22,31 @@
     "multi select"
   ],
   "license": "MIT",
-  "peerDependencies": {
```
Comment (sommmen) [inline package.json:15]: Seems you removed the peer dependencies - is that o.k?
The angular packages are defined as dev deps - but i can imagine you'd need those 3 packages at runtime.

again, no js dev so not sure.

**Review by sommmen — COMMENTED** [review]
LGTM!

**Comment by sommmen** [comment]
> @sommmen FYI: if you don't want to build a local version yourself, your can always use a published version from https://www.npmjs.com/package/ngx-bootstrap-multiselect-updated
> 
> The fork/PR I created is based on that NPM package (which is also a fork from the original ngx-bootstrap-multiselect from @softsimon). I hope this helps.

Hiya - thanks for sharing your knowledge.
I forget i'm still on Angular 14, so i'd have to upgrade Angular in the future.

File: package.json (hunk 1 above)
Comment (dsteegen) [reply package.json:15]: That's a valid comment! It's indeed better to specify the peer dependencies to make sure that the project using this library is compatible with the specified dependencies. I will try to add those this evening or tomorrow :)

**Comment by dsteegen** [comment]
> > @sommmen FYI: if you don't want to build a local version yourself, your can always use a published version from https://www.npmjs.com/package/ngx-bootstrap-multiselect-updated
> > The fork/PR I created is based on that NPM package (which is also a fork from the original ngx-bootstrap-multiselect from @softsimon). I hope this helps.
> 
> Hiya - thanks for sharing your knowledge. I forget i'm still on Angular 14, so i'd have to upgrade Angular in the future.

Normally you would still be able to use this library with Angular 14 since there are no breaking API changes compared to version 15.

**Comment by sommmen** [comment]
@softsimon don't forget to create a release ❤️

**Comment by dsteegen** [comment]
@softsimon if there is anything we do from our end to smoothen the release, just let us know. If not, I appreciate the time and effort you are putting into this 🙂

**Comment by softsimon** [comment]
@sommmen 
@dsteegen 

```
npm notice Publishing to https://registry.npmjs.org/
+ ngx-bootstrap-multiselect@4.0.0
```

**Comment by dsteegen** [comment]
Cheers! Thank you for taking care of this.

**Pull request merged**